config/yaml-mp/src/main/java/io/helidon/config/yaml/mp/YamlConfigSourceProvider.java

```diff
@@ -21,30 +21,47 @@
 import java.util.Enumeration;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
 
 import io.helidon.config.ConfigException;
+import io.helidon.config.mp.spi.MpConfigSourceProvider;
 
 import org.eclipse.microprofile.config.spi.ConfigSource;
-import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
 
 /**
  * YAML config source provider for MicroProfile config that supports file {@code application.yaml}.
  * This class should not be used directly - it is loaded automatically by Java service loader.
  */
-public class YamlConfigSourceProvider implements ConfigSourceProvider {
+public class YamlConfigSourceProvider implements MpConfigSourceProvider {
+    private static final String RESOURCE_NAME = "application.yaml";
+
     @Override
     public Iterable<ConfigSource> getConfigSources(ClassLoader classLoader) {
-        Enumeration<URL> resources;
-        try {
-            resources = classLoader.getResources("application.yaml");
-        } catch (IOException e) {
-            throw new ConfigException("Failed to read resources from classpath", e);
-        }
+        return getConfigSources(classLoader, Optional.empty());
+    }
+
+    @Override
+    public Iterable<ConfigSource> getConfigSources(ClassLoader classLoader, String profile) {
+        Objects.requireNonNull(profile, "Profile must not be null");
+        return getConfigSources(classLoader, Optional.of(profile));
+    }
 
+    private Iterable<ConfigSource> getConfigSources(ClassLoader classLoader, Optional<String> profile) {
```

Comment:
For private methods, you can use `null`, the restriction is for public API.
Also optional parameters are not good.

Reply:
Will take note of avoiding optional from hereon. Private methods were refactored to allow for null parameter.